pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -180,8 +198,12 @@ func NewController(client kubernetes.Interface, options Options) *Controller {
 	svcInformer := sharedInformers.Core().V1().Services().Informer()
 	out.services = out.createCacheHandler(svcInformer, "Services")
 
-	epInformer := sharedInformers.Core().V1().Endpoints().Informer()
-	out.endpoints = out.createEDSCacheHandler(epInformer, "Endpoints")
+	switch options.EndpointMode {
+	case EndpointsOnly:
+		out.endpoints = newEndpointsController(out, sharedInformers)
+	case EndpointSliceOnly:
+		out.endpoints = newEndpointSliceController(out, sharedInformers)
```

Comment:
default case?

Reply:
Ah yeah, used to work in a language that asserted at compile time that switches had all possible cases covered.. I'll add one here